kratos/utilities/read_materials_utility.cpp

```diff
@@ -0,0 +1,265 @@
+    //if (data["Material"]["Variables"].end() - data["Material"]["Variables"].begin())
+    //    KRATOS_INFO("::[Reading materials process DEBUG]::")
+    //            << "Property " << property_id << " is not empty." << std::endl;
+    if (p_prop->HasVariables())
+        KRATOS_INFO("Read materials")
+            << "Property " << property_id << " already has variables." << std::endl;
+    //if (len(data["Material"]["Tables"].keys()) > 0 && p_prop.HasTables())
+    if (p_prop->HasTables())
+        KRATOS_INFO("Read materials")
+            << "Property " << property_id << " already has tables." << std::endl;
+    */
+
+    // Assign the p_properties to the model part's elements and conditions.
+    auto& elements_array = model_part.Elements();
+    auto& conditions_array = model_part.Conditions();
+
+    #pragma omp parallel for
+    for(int i = 0; i < static_cast<int>(elements_array.size()); ++i) {
+        auto it_elem = elements_array.begin() + i;
+        it_elem->SetProperties(p_prop);
+    }
+
+    #pragma omp parallel for
+    for(int i = 0; i < static_cast<int>(conditions_array.size()); ++i) {
+        auto it_cond = conditions_array.begin() + i;
+        it_cond->SetProperties(p_prop);
+    }
+
+    //Set the CONSTITUTIVE_LAW for the current p_properties.
+    if (data["Material"].Has("constitutive_law")) {
+        std::string constitutive_law_name = data["Material"]["constitutive_law"]["name"].GetString();
+
+        // Remove application info from consitutive law name.
+        // Ex: KratosMultiphysics.StructuralMechanicsApplication.LinearElastic3D -> LinearElastic3D
+        constitutive_law_name = CleanVariableName(constitutive_law_name);
+
+        auto p_constitutive_law = KratosComponents<ConstitutiveLaw>().Get(constitutive_law_name).Clone();
+        p_prop->SetValue(CONSTITUTIVE_LAW, p_constitutive_law);
+    } else {
+        KRATOS_INFO("Read materials") << "Not consitutive law defined for material ID: " << property_id << std::endl;
+    }
+
+    // Add / override the values of material parameters in the p_properties
+    Parameters variables = data["Material"]["Variables"];
+    for(auto iter = variables.begin(); iter != variables.end(); iter++) {
+        const Parameters value = variables.GetValue(iter.name());
+
+        // Remove application info from variable name.
+        // Ex: KratosMultiphysics.YOUNG_MODULUS -> YOUNG_MODULUS
+        const std::string& variable_name = CleanVariableName(iter.name());
+
+        // We don't just copy the values, we do some tyransformation depending of the destination variable
+        if(KratosComponents<Variable<double> >::Has(variable_name)) {
+            const Variable<double>& variable = KratosComponents<Variable<double>>().Get(variable_name);
+            if (value.IsDouble()) {
+                p_prop->SetValue(variable, value.GetDouble());
+            } else if (value.IsInt()) {
+                p_prop->SetValue(variable, static_cast<double>(value.GetInt()));
```

Comment:
I think these transformations are very dangerous!

silently changing is really not a good idea IMO

the user should specify the correct type

Reply:
Well, this was a @RiccardoRossi  suggestion

Reply:
I will not block bcs of this, but I really don't think we should give users opportunities to become sloppy with input
Who knows which problems this can create down the road...

Your call, just keep it in mind

Reply:
I understand that this is for read` young_modulus = 1000000` without getting an error. Isn't it?

Reply:
You are repeating this ifs in many places. It would be better to add a method ExtractDouble y mover todo ahi.

Reply:
if you do 
     
       param.GetDouble()

and it is a "number" than it should convert it transparently, doing the casting internally.

@philbucher i think we should accept that if someone does not put the . we can still convert integer to double...

Reply:
Where is this conversion done?

https://github.com/KratosMultiphysics/Kratos/blob/8218d2689a6d2a55bb52b33c8ca7a80091d62d34/kratos/includes/kratos_parameters.h#L330